lib/std/private/bitops_utils.nim

```diff
@@ -10,8 +10,23 @@ template forwardImpl*(impl, arg) {.dirty.} =
     else:
       impl(x.uint64)
 
-template toUnsigned*(x: int8): uint8 = cast[uint8](x)
-template toUnsigned*(x: int16): uint16 = cast[uint16](x)
-template toUnsigned*(x: int32): uint32 = cast[uint32](x)
-template toUnsigned*(x: int64): uint64 = cast[uint64](x)
-template toUnsigned*(x: int): uint = cast[uint](x)
+template toUnsigned(T: typedesc[SomeInteger and not range]): untyped =
+  # copied/adapted from std/typetraits.toUnsigned to avoid a `system => typetraits` dependency
```

Comment:
If you cannot use typetraits here, there is no need to copy&paste, you can keep the old version of bitops_utils.nim as it's at least shorter.

Reply:
done, but kept the name castToUnsigned as toUnsigned is misleading (it casts instead of doing a conversion in traditional sense); also the way I'm doing it allows simplifying all call sites by avoiding to branch on whether the input is signed vs unsigned